index.js

```diff
@@ -0,0 +1,145 @@
+      as: 'director'
+    }]
+  })
+
+  if (matchingmovie.length) {
+    response.send(matchingmovie)
+  } else {
+    response.sendStatus(404)
+  }
+})
+
+app.get('/director/:identifier', async (request, response) => {
+  const { identifier } = request.params
+  const matchingdirector = await models.director.findAll({
+    where: {
+      [Op.or]: [{ id: identifier }]
+    },
+    include: [{
+      model: models.movie,
+      as: 'movie'
+    }]
+  })
+
+  if (matchingdirector.length) {
+    response.send(matchingdirector)
+  } else {
+    response.sendStatus(404)
+  }
+})
+app.get('/genre/:gname', async (request, response) => {
+  const { gname } = request.params
+  const matchinggenre = await models.genre.findAll({
+    where: {
+      [Op.or]: [{ name: gname }]
+    },
+    include: [{
+      model: models.movie,
+      as: 'movie'
+    }]
+  })
+
+  if (matchinggenre.length) {
+    response.send(matchinggenre)
+  } else {
+    response.sendStatus(404)
+  }
+})
+app.delete('/movie/:identifier', async (request, response) => {
+  const { identifier } = request.params
+  const matchingmovie = await models.movie.findOne({
+    where: {
+      [Op.or]: [{ id: identifier }]
+    },
+    include: [{
+      model: models.genre,
+      as: 'genre'
+    }, {
+      model: models.director,
+      as: 'director'
+    }]
+  })
+
+  const movieGenres = matchingmovie.getGenre()
+  const movieDirectors = matchingmovie.getDirector()
+  matchingmovie.removeGenre(movieGenres)
+  matchingmovie.removeDirector(movieDirectors)
+  await models.movie.destroy({
+    where: { id: identifier }
+  })
+
+  // If everything goes well respond with the movie
+  return response.status(200).send('movie deleted successfully')
+})
+app.use(bodyParser.json())
+app.post('/movie', async (request, response) => {
+  const { title, director, releaseDate, rated, runTime, genre } = request.body
+  if (!title || !director || !releaseDate || !rated || !runTime || !genre) {
+    response.status(400).send('The following attributes are required: title, director, releaseDate, rated, runTime, genre')
+  }
+
+  const existingMovie = await models.movie.findOne({ where: { title: title } })
+  if (!existingMovie) {
+    const movie = await models.movie.create({ title: title, rated: rated, runTime: runTime, releaseDate: releaseDate })
+    const genreArr = genre.split(',')
+    const genreIds = []
+
+    for (var i = 0; i < genreArr.length; i++) {
+      const genre = await models.genre.findOrCreate({ where: { name: genreArr[i] } })
+      genreIds.push(genre.id)
+    }
+
+    const directorArr = director.split(',')
+    const directorIds = []
+
+    for (var i = 0; i < directorArr.length; i++) {
+      const director = await models.director.findOrCreate({ where: { name: directorArr[i] } })
+      directorIds.push(director.id)
+    }
+
+    movie.setGenre(genreIds)
+    movie.setDirector(directorIds)
+    movie.save()
+    response.status(200).send('movie created sucessfully')
```

Comment:
Nice job on this, to make it cleaner, look into Sequelize's `findOrCreate` function.